dts/bindings/mtd/jedec,mspi-nor.yaml

```diff
@@ -30,3 +32,17 @@ properties:
       Maximum time, in milliseconds, allowed for a single transfer on the MSPI
       bus in communication with the flash chip. The default value is the one
       that was previously hard-coded in the flash_mspi_nor driver.
+
+  use-4byte-addressing:
+    type: boolean
+    description: |
+      Indicates that 4-byte addressing is to be used in communication with
+      the flash chip. The driver will use dedicated instruction codes for
+      commands that require addresses (like Read, Page Program, or Erase)
+      and will switch the flash chip to 4-byte addressing mode if necessary.
+
```

Comment:
switching the flash chip to 4-byte addressing mode, is only necessary when the regular instruction codes and not the 4-byte specific ones are used. Some SoCs and FPGAs still rely that the regular instruction codes are for 3 byte addresses.

There is also already a `use-4b-addr-opcodes` prop for that in `jedec,spi-nor-common.yaml`, that this includes.

Reply:
I modified the `ENTER_4BYTE_ADDR()` macro so that now no switching is done if the chip supports the special 4-byte address instruction codes: https://github.com/zephyrproject-rtos/zephyr/blob/6af1921305baa350882592683346f5f08ddcea65/drivers/flash/flash_mspi_nor_sfdp.h#L302-L306
I also rephrased the above description. It should now more accurately tell what the driver does.